app/code/Magento/CatalogInventory/Api/InventoryRegistryInterface.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Magento\CatalogInventory\Api;
+
+use Magento\CatalogInventory\Api\Data\InventoryInterface;
+use Magento\CatalogInventory\Api\Data\InventoryRecordInterface;
+
+/**
+ * Inventory registry interface
+ *
+ * The main point, where inventory information is retrieved for Magento
+ *
+ * @api
+ */
+interface InventoryRegistryInterface
+{
+    /**
+     * Returns main store inventory based on store configuration
+     * or other environment dependencies
+     *
+     * This one should be used only for filtering stock status on collections,
+     * used in indexers for applying inventory status limitations
+     *
+     * @param int|null $storeId
+     * @return InventoryInterface
+     */
+    public function getInventory($storeId = null);
```

Comment:
I believe we want to eliminate dependency on store id and introduce some "virtual" scope identifier which might be configured per instance. Have you considered such option?